app/controllers/mainstream_browse_pages_controller.rb

```diff
@@ -1,46 +1,84 @@
 class MainstreamBrowsePagesController < ApplicationController
-  include TagCreateUpdatePublish
-  tag_create_update_publish_for MainstreamBrowsePage
-
   before_filter :require_gds_editor_permissions!
 
-  def edit
-    set_topics_for_select
+  def index
+    @pages = MainstreamBrowsePage.sorted_parents
   end
 
-  def show; end
+  def show
+    @browse_page = find_browse_page
+  end
+
+  def edit
+    @browse_page = find_browse_page
+    @topics_for_select = topics_for_select
+  end
 
   def update
-    # Convert the String ids to Topic objects so that
-    # `update_attributes` correct updates and associates
-    # them with the given `MainstreamBrowsePage` object.
-    if params.require(:mainstream_browse_page).key? :topics
-      topic_ids = params.require(:mainstream_browse_page)[:topics]
-      topics = topic_ids.reject(&:blank?).map { |t| Topic.find(t) }
-      msb_params = tag_params.merge({"topics" => topics})
+    browse_page = find_browse_page
+
+    if browse_page.update_attributes(browse_page_params)
+      PanopticonNotifier.update_tag(MainstreamBrowsePagePresenter.new(browse_page))
+      PublishingAPINotifier.send_to_publishing_api(browse_page)
+      redirect_to browse_page
     else
-      msb_params = tag_params
+      @topics_for_select = topics_for_select
+      render :edit
     end
+  end
+
+  def new
+    @browse_page = MainstreamBrowsePage.new(parent_id: params[:parent_id])
+  end
 
-    if @resource.update_attributes(msb_params)
-      PanopticonNotifier.update_tag(
-        presenter_klass.new(@resource)
-      )
-      PublishingAPINotifier.send_to_publishing_api(@resource)
+  def create
+    browse_page = MainstreamBrowsePage.new
+    browse_page.attributes = tag_params
 
-      redirect_to mainstream_browse_page_path(@resource)
+    if browse_page.save
+      PanopticonNotifier.create_tag(MainstreamBrowsePagePresenter.new(browse_page))
+      PublishingAPINotifier.send_to_publishing_api(browse_page)
+      redirect_to browse_page
     else
-      set_topics_for_select
-      render :edit
+      @browse_page = browse_page
+      render 'new'
     end
   end
 
+  def publish
+    browse_page = find_browse_page
+
+    browse_page.publish!
+    PanopticonNotifier.publish_tag(MainstreamBrowsePagePresenter.new(browse_page))
+    PublishingAPINotifier.send_to_publishing_api(browse_page)
+    redirect_to browse_page
+  end
+
 private
-  def presenter_klass
-    MainstreamBrowsePagePresenter
+
+  def tag_params
+    params.require(:mainstream_browse_page)
+      .permit(:slug, :title, :description, :parent_id)
+  end
```

Comment:
Is there any reason to keep this method separate from `browse_page_params` any more?